src/QsCompiler/Transformations/SyntaxTreeTrimming.cs

```diff
@@ -15,22 +13,18 @@ namespace Microsoft.Quantum.QsCompiler.Transformations.SyntaxTreeTrimming
 {
     public static class TrimSyntaxTree
     {
-        public static QsCompilation Apply(QsCompilation compilation, bool keepAllIntrinsics)
+        public static QsCompilation Apply(QsCompilation compilation, bool keepAllIntrinsics, IEnumerable<QsQualifiedName>? dependencies = null)
         {
-            return TrimTree.Apply(compilation, keepAllIntrinsics);
+            return TrimTree.Apply(compilation, keepAllIntrinsics, dependencies);
         }
 
         private class TrimTree : SyntaxTreeTransformation<TrimTree.TransformationState>
         {
-            public static QsCompilation Apply(QsCompilation compilation, bool keepAllIntrinsics)
+            public static QsCompilation Apply(QsCompilation compilation, bool keepAllIntrinsics, IEnumerable<QsQualifiedName>? dependencies)
             {
                 var globals = compilation.Namespaces.GlobalCallableResolutions();
-                var augmentedEntryPoints = BuiltIn.AllBuiltIns
-                    .Where(bi => bi.Kind != BuiltInKind.Attribute && globals.ContainsKey(bi.FullName))
-                    .Select(bi => bi.FullName)
-                    .Concat(compilation.EntryPoints)
-                    .Distinct()
-                    .ToImmutableArray();
+                var dependenciesToKeep = dependencies?.Where(globals.ContainsKey) ?? ImmutableArray<QsQualifiedName>.Empty;
+                var augmentedEntryPoints = dependenciesToKeep.Concat(compilation.EntryPoints).Distinct().ToImmutableArray();
```

Comment:
nit: A more typical syntax when using multiple LINQ methods is to have each on a new line. This reflects SQL syntax.

Reply:
done.